src/marqo/index.py

```diff
@@ -52,19 +52,28 @@ def __init__(
         self.updated_at = self._maybe_datetime(updated_at)
 
         skip_version_check = False
+        # trying to get index url to verify that index is mapped
         try:
             self.config.instance_mapping.get_index_base_url(self.index_name)
         except errors.MarqoError as e:
-            mq_logger.warning(e)
+            mq_logger.debug(
+                f'Cache update on index object instantiation could not retrieve index URL: {e}')
             skip_version_check = True
-        if not skip_version_check:
+
+        if (self.config.instance_mapping.is_index_usage_allowed(index_name=self.index_name)
```

Comment:
isn't `is_index_usage_allowed making` the same as `            self.config.instance_mapping.get_index_base_url(self.index_name)
` above on line 57?

Reply:
This is because now `get_index_base_url` doesn't care whether an index is CREATING or READY. `is_index_usage_allowed`, on the other hand, does care.